### PR TITLE
Automatically deploy to CDF on master

### LINKED ISSUE #49: Deployment to CDF - Stale javascript files
CDF deployment isn't working as intended. 

`HTML` files are updated while `javascript` files are being ignored. 

Redeployment will indicate that `"Everything is up to date"`. 

http://draft.cssu.ca is working fine.

### PR DESCRIPTION
**Pull request by elliottsj**
This is simpler than our current process of deploying to CDF manually via git. With these changes, any commits pushed to master (including merging draft into master) will automatically deploy to CDF.

This works by having Travis CI build the static site, then copy it to dbsrv1 via rsync using a deploy key I created using `ssh-keygen`. The private key is encrypted via https://docs.travis-ci.com/user/encrypting-files/; on the master branch, Travis CI will decrypt it and use it to connect to dbsrv1.

Resolves #49, #51.

### CODE REVIEW

**Comment by g3wanghc** [comment]
Looks good. 👍 

Thanks a bunch @elliotts! Hopefully this will resolve issues #49 and #51.

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+eval "$(ssh-agent -s)"
+chmod 600 .travis/deploy_key
+ssh-add .travis/deploy_key
+
+# Copy the static site to CDF, preserving the gallery and wiki directories
+rsync \
+  --archive \
+  --compress \
+  --delete \
+  --exclude="/gallery/" \
```
Comment (arkon) [inline deploy.sh:24]: I think there were a lot more directories on CDF, weren't there? I remember there being a bunch of junk, but I could be wrong.

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+# Strict mode: http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+# Add dbsrv1 to known_hosts
+cat << EOF >> ~/.ssh/known_hosts
+dbsrv1.cdf.toronto.edu ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAIEAydP6bdmyTALluMTn4xEs+Fu8OAFDbyV93owVVNGxerSvNijXRt7BuvaCY13mXtCyJ2BsDzNYZ5kR/+FFCIv5e6/3Y5W4yjkwRZ5/Drt5ZK2fJtmeRYyVOztQHSyWIY9KBJmrcGpDPpK6LY6kseR2djWBfwGLCejSKXWsZ5+Oz/0=
+EOF
```
Comment (arkon) [inline deploy.sh:9]: Is this to handle some sort of user input?

File: deploy.sh (hunk 1 above)
Comment (elliottsj) [inline deploy.sh:24]: Here's what we have in `/space/data/www/cssu/htdocs`:

```
LICENSE
Rakefile
about
assets
autoprefixer.yml
awards
browserconfig.xml
downloads
events
feed.xml
fonts
frosh_volunteer
gallery
humans.txt
index.html
news
robots.txt
sponsors
store
tag
w
```

And here's the generated `_site`:

```
Gemfile.lock
LICENSE
Procfile
Rakefile
about
assets
autoprefixer.yml
awards
browserconfig.xml
downloads
events
feed.xml
fonts
frosh_volunteer
humans.txt
index.html
news
robots.txt
sponsors
store
tag
```

Seem CDF only has extras `gallery` and `w`. The generated site also has `Gemfile.lock` and `Procfile`; we should configure jekyll to exclude those from the generated site

File: deploy.sh (hunk 2 above)
Comment (elliottsj) [inline deploy.sh:9]: It's a way to do 'heredoc'-style strings in bash: http://stackoverflow.com/questions/2500436/how-does-cat-eof-work-in-bash

So we don't have a super long `echo "..." >> ~/.ssh/known_hosts` statement

File: deploy.sh (hunk 1 above)
Comment (arkon) [inline deploy.sh:24]: Alright, I must have been thinking of something else then.

**Comment by arkon** [comment]
LGTM.

**Pull request merged**